docs/docs/specs/v0/composer.md

```diff
@@ -0,0 +1,91 @@
+# Composer
+
+A developer-first, peer-to-peer knowledge management system.
+
+## Scenarios
+
+1. Users can collaborate on a github issue together
+2. Users can work on folders of markdown files
+3. Developers can extend Composer with custom data sources and visual surfaces
+
+## Extensibility
+> "Composer Chrome" refers to the total composer UI organization and all affordances surrounding the content the users interact with - all UI which is not the content itself.
+
+Two levels of complexity are proposed in sequence:
+
+**Static chrome**, where we constrain the application chrome to be statically defined and not extensible by plugins. Elements of the chrome may each expose some API surface for plugins to extend, but the overall elements of the chrome are fixed and can be assumed by all plugins without explicit dependency expressions.
+
+**Dynamic chrome**, where the chrome and it's organization is also extensible and now plugins have to express explicitly the elements of chrome they depend on because their presence can no longer be assumed.
+
+These are in addition to another orthogonal dimension of complexity, the way we load plugins:
+
+**Static plugins**, where a `<Composer />` element is given a list of `Plugin[]` directly as props and has to be compiled into the app at build time.
+
+**Dynamic plugins**, where the app knows how to load plugins via ESM `import` calls in the browser at runtime.
```

Comment:
It makes sense to me that composer could provide a plugin interface more specific to it's chrome structure. We would then support those plugins being dynamically loaded/installed at some point. What I'm struggling with right now is reconciling this with the frames concept.

Plugins can include frames they require for rendering, this seems fine, but is still composer-specific. If frames are going to be published independent of composer plugins, then frames need their own system for being loaded/managed in some way that's not specific to composer. If that is true, then are we going to build the same thing again expect for plugins or would plugins somehow use that more generic loading system?

I feel like we want to work backwards from a high-level vision of the dynamic end-state because changes to the goal will likely influence where we start and make it easier/harder to get there.

Reply:
> Plugins can include frames they require for rendering, this seems fine, but is still composer-specific
yes composer specific is the design intention for the sake of reducing complexity

> If frames are going to be published independent of composer plugins
I don't think we should worry about that for now (seems like a story for sharing frames between more than one app)

Reply:
its good to work backwards from a very high level north star, but that doesn't mean we need to pre-engineer for all the complexity of that world today - the opposite. We need to work towards increasing complexity from a simpler place, accepting that we will go through major refactoring and major incompatibility boundaries on the way there. The problem with pre-engineering for more complexity too early is that it still doesn't eliminate these major refactors and compatibility boundaries, (because practice will teach us what's actually important and what's not) - but having more complexity from day 1 means more to worry about during these refactors, and more to throw away.

Reply:
> yes composer specific is the design intention for the sake of reducing complexity

I didn't mean to imply it being composer-specific was bad, but just that it imposes some level of dependency restrictions (i.e., dxos deps don't know about this, it's a braneframe/composer-level thing)

Reply:
Fair enough about not pre-engineering things which might just end up in the trash. I think what makes me uneasy is having this frames concept that we've discussed quite a bit, have the expectation that it will end up under-girding plugins someway in the long run, but not have anything specifying that (even if it's fuzzy/uncertain/likely will change shape).

Reply:
addd a non goals section for debate